Course/Course/Presentation/Dates/CourseDatesView.swift

```diff
@@ -75,134 +75,208 @@ struct Line: Shape {
+    let courseDateBlockDict: [Date: [CourseDateBlock]]
+    let viewModel: CourseDatesViewModel
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 5) {
+            // Toggle button to expand/collapse the cell
+            Button(action: {
+                withAnimation {
+                    isExpanded.toggle()
+                }
+            }) {
+                HStack {
+                    VStack(alignment: .leading) {
+                        Text(CompletionStatus.completed.rawValue)
+                            .font(Theme.Fonts.titleSmall)
+                            .foregroundColor(Theme.Colors.textPrimary)
+                        
+                        if !isExpanded {
+                            let totalCount = courseDateBlockDict.values.reduce(0) { $0 + $1.count }
+                            Text("\(totalCount) \(CoreLocalization.CourseDates.itemsHidden)")
+                                .font(Theme.Fonts.labelMedium)
+                                .foregroundColor(Theme.Colors.textPrimary)
+                        }
+                    }
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .padding(.leading, 16)
+                    .padding(.vertical, 8)
+                    
+                    Image(systemName: "chevron.down")
+                        .labelStyle(.iconOnly)
+                        .rotationEffect(.degrees(isExpanded ? 180 : 0))
+                        .foregroundColor(Theme.Colors.textPrimary)
+                        .padding()
+                }
+            }
+            
+            // Your expandable content goes here
+            if isExpanded {
+                VStack(alignment: .leading) {
+                    ForEach(courseDateBlockDict.keys.sorted(), id: \.self) { date in
+                        let blocks = courseDateBlockDict[date]!
+                        let block = blocks[0]
+                        
+                        Spacer()
+                        Text(block.formattedDate)
+                            .font(Theme.Fonts.labelMedium)
+                            .foregroundStyle(Theme.Colors.textPrimary)
+                        
+                        ForEach(blocks) { block in
+                            HStack(alignment: .top) {
+                                block.blockImage?.swiftUIImage
+                                    .foregroundColor(Theme.Colors.textPrimary)
+                                StyleBlock(block: block, viewModel: viewModel)
+                                    .padding(.bottom, 15)
+                                Spacer()
+                                if block.canShowLink && !block.firstComponentBlockID.isEmpty {
+                                    Image(systemName: "chevron.right")
+                                        .resizable()
+                                        .scaledToFit()
+                                        .frame(width: 6.55, height: 11.15)
```

Comment:
Just curious, how do you calculate that kind of accuracy? :)

Reply:
I got this precision from [figma](https://www.figma.com/file/iZ56YMjbRMShCCDxqrqRrR/Open-edX-Mobile-App-All-Screens-v2.0?type=design&node-id=7308-69673&mode=design&t=oZAQ68nOf3nYvST7-4) designs 😄 
<img width="602" alt="Screenshot 2024-01-29 at 1 07 08 PM" src="https://github.com/openedx/openedx-app-ios/assets/11990137/5bbee691-4c85-4f47-80c6-99a4e8693471">